src/datetime.js

```diff
@@ -199,16 +206,52 @@ function toTechTimeFormat(
     includeOffset,
     includeZone = false,
     spaceZone = false,
-    format = "extended"
+    format = "extended",
+    precision = "milliseconds"
   }
 ) {
-  let fmt = format === "basic" ? "HHmm" : "HH:mm";
-
-  if (!suppressSeconds || dt.second !== 0 || dt.millisecond !== 0) {
-    fmt += format === "basic" ? "ss" : ":ss";
-    if (!suppressMilliseconds || dt.millisecond !== 0) {
-      fmt += ".SSS";
-    }
+  // validate the precision and throw if invalid
+  let desiredPrecision = Precision[Duration.normalizeUnit(precision)];
```

Comment:
Wondering if this needs to call the static `normalizeUnit` on `Duration` vs local to datetime.js?